marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/triggers/mapping/mappingJSONtoXML.sjs

```diff
@@ -24,6 +24,7 @@ if (esMappingLib.versionIsCompatibleWithES()) {
   xdmp.invokeFunction(function () {
       const es = require('/MarkLogic/entity-services/entity-services');
       es.mappingPut(xmlURI);
+      xdmp.nodeDelete(cts.doc(xmlURI+'.xslt').xpath('//xdmp:import-module[@href="/MarkLogic/entity-services/standard-library.xqy"]'));
```

Comment:
Does this work?? I'm surprised it would, because the xmlURI doc doesn't exist until the es.mappingPut call commits.

Reply:
It works, I have added a test to ensure that it gets removed.

Reply:
I guess es.mappingPut must be running in a separation transaction then?? Otherwise the ".xslt" file wouldn't exist yet when the nodeDelete is performed.

Reply:
es.mappingPut does an eval to insert the xslt, so I think the doc insert is committed before the node is deleted